### DISCUSSION TITLE
is there a git add remote extension

### DISCUSSION THREAD

**Discussion by nick5454** [cake-build/cake#3416] [Extension Q&A] [answered]
I need to add a remote repository in my cake build.

`The git line is "git remote add <name> <remote url>"`

I looked at this package [Cake.Git](https://cakebuild.net/dsl/git/)

But I don't see that command in there. Is there a package that does that or allows me to issue custom commands in git?

**Comment by nick5454** [comment]
I can just do a command line process, it looks like

**Comment by augustoproiete** [comment, marked as answer]
[Cake.Git](https://github.com/cake-contrib/Cake_Git) is a wrapper around [LibGit2Sharp](https://github.com/libgit2/libgit2sharp), so in the case of remotes, it's a feature that LibGit2Sharp supports but it's not yet exposed as a Cake alias in the Cake.Git addin. Thus you can call into LibGit2Sharp directly yourself:

```cakescript
#addin "nuget:?package=Cake.Git&version=1.0.1"

using LibGit2Sharp;

using (var repository = new Repository(@"C:\augusto\my-git-repo"))
{
    var remoteUrl = "https://github.com/owner/my-git-repo.git";
    repository.Network.Remotes.Update("origin", r => { r.Url = remoteUrl; });
}
```
